mlflow/models/evaluation/base.py

```diff
@@ -854,38 +854,45 @@ def __init__(self, endpoint, params):
             self.endpoint = endpoint
             self.params = params
 
-        def predict(self, context, model_input: pd.DataFrame):
-            if len(model_input.columns) != 1:
-                raise MlflowException(
-                    f"The number of input columns must be 1, but got {model_input.columns}. "
-                    "Multi-column input is not supported for evaluating an MLflow Deployments "
-                    "endpoint. Please include the input text or payload in a single column.",
-                    error_code=INVALID_PARAMETER_VALUE,
-                )
-            input_column = model_input.columns[0]
-
-            predictions = []
-            for data in model_input[input_column]:
-                if isinstance(data, str):
-                    # If the input data is a string, we will construct the request payload from it.
-                    prediction = _call_deployments_api(self.endpoint, data, self.params)
-                elif isinstance(data, dict):
-                    # If the input data is a dictionary, we will directly use it as the request
-                    # payload, with adding the inference parameters if provided.
-                    prediction = _call_deployments_api(
-                        self.endpoint, data, self.params, wrap_payload=False
-                    )
-                else:
+        def predict(self, context, model_input: Union[Dict, pd.DataFrame]):
+            # The model input is typically a Pandas DataFrame that contains the
+            # request payloads in a single column.
+            # However, if the model_type is "databricks-agents" and the Databricks
+            # evaluator is used, this PythonModel is invoked with a dictionary
+            # corresponding to the ChatCompletionsRequest schema.
+            if isinstance(model_input, dict):
+                return self._predict_single(model_input)
```

Comment:
Can we also handle the case where we have a list of dicts, e.g. a batch of chat histories?

Reply:
@dbczumar Could you elaborate what "a batch of chat histories" means? The RAG evaluator does not send a batch request to the model. Or will it be for future enhancement?

Reply:
Yeah, future enhancement. If it's easy, can we future proof?

Reply:
Yea it's easy to implement, I will add it with that note in comments.